hbase-http/src/test/java/org/apache/hadoop/hbase/http/TestHttpServer.java

```diff
@@ -269,6 +277,35 @@ public void testContentTypes() throws Exception {
     // assertEquals("text/html; charset=utf-8", conn.getContentType());
   }
 
+  @Test
+  public void testNegotiatesEncodingGzip() throws IOException {
+    try (final CloseableHttpClient client = HttpClients.createMinimal()) {
+      final HttpGet request = new HttpGet(new URL(baseUrl, "/static/test.css").toString());
+
+      request.setHeader(HttpHeaders.ACCEPT_ENCODING, null);
+      final long unencodedContentLength;
+      try (final CloseableHttpResponse response = client.execute(request)) {
+        final HttpEntity entity = response.getEntity();
+        assertNotNull(entity);
+        assertNull(entity.getContentEncoding());
+        unencodedContentLength = entity.getContentLength();
+        MatcherAssert.assertThat(unencodedContentLength, greaterThan(0L));
+      }
+
+      request.setHeader(HttpHeaders.ACCEPT_ENCODING, "gzip");
+      final long encodedContentLength;
+      try (final CloseableHttpResponse response = client.execute(request)) {
+        final HttpEntity entity = response.getEntity();
+        assertNotNull(entity);
+        assertNotNull(entity.getContentEncoding());
+        assertEquals("gzip", entity.getContentEncoding().getValue());
+        encodedContentLength = entity.getContentLength();
+        MatcherAssert.assertThat(encodedContentLength, greaterThan(0L));
+      }
+      MatcherAssert.assertThat(unencodedContentLength, greaterThanOrEqualTo(encodedContentLength));
```

Comment:
I think we'd better use greaterThan here? If we do not enable gzip, the content length will be the same, so the the asserttion here just does nothing...

Reply:
What about using the minimal HTTP client with `Accept-Encoding: gzip`, and then validate that we can uncompress the data successfully?

Reply:
I could do, sure. The goal is to test that the server honors the encoding request, not the efficiency of gzip for this specific content.

Reply:
I could do, sure. The goal is to test that the server honors the encoding request, not that the httpcomponents library handles gzip streams correctly.